metecho/api/models.py

```diff
@@ -554,8 +557,13 @@ def should_update_in_progress(self):
 
     def should_update_review(self):
         task_statuses = self.tasks.values_list("status", flat=True)
-        return task_statuses and all(
-            status == TASK_STATUSES.Completed for status in task_statuses
+        return (
+            task_statuses
+            and all(
+                status in [TASK_STATUSES.Completed, TASK_STATUSES.Canceled]
+                for status in task_statuses
+            )
+            and any(status == TASK_STATUSES.Completed for status in task_statuses)
```

Comment:
> @jerivas There's probably a more efficient way to do this. The goal is that this should return truthy if all tasks are completed or cancelled, but not all tasks are cancelled.

I like it this way, one query and all conditions are clear and readable. We can probably do it with different `filter()` and `count()` calls but I don't think it would be more efficient or readable that the current implementation. I'd just add `return truthy if all tasks are completed or cancelled, but not all tasks are cancelled` as a docstring (GitHub won't let me do it as a suggestion)

Reply:
Added in: https://github.com/oddbird/Metecho/commit/6c678a89e9b3975cfd7d6232c0cd62b933dc4309